cmd/libsnap-confine-private/string-utils.c

```diff
@@ -69,3 +70,48 @@ int sc_must_snprintf(char *str, size_t size, const char *format, ...)
 
 	return n;
 }
+
+char *sc_must_stpcpy(char *buf, size_t buf_size, char *dest, const char *src)
+{
+	// Set errno in case we die.
+	errno = 0;
+	if (buf == NULL) {
+		die("cannot append string: buffer is NULL");
+	}
+	if (dest == NULL) {
+		die("cannot append string: destination is NULL");
+	}
+	if (src == NULL) {
+		die("cannot append string: source is NULL");
+	}
+	// Sanity check, the code doesn't need buffers larger than a few KBs so
+	// prevent corrupted or otherwise huge buffers from seeming "valid".
+	if (buf_size >= 0xFFFF) {
```

Comment:
Because we don't need big buffers and our protections break down (note that overflow detection is not implemented to guard against integer overflow). Limiting the maximum buffer size to something small (and entirely reasonable for where we want to use this) adds some security.